README.md

```diff
@@ -122,58 +127,52 @@ provider classes, and not directly call into `CommandReference` or `Node`.
 
 These utilities can be used directly on a Cisco device (as used by Puppet
 and Chef) or can run on a workstation and point to a Cisco device (as used
-by the included minitest suite).
-
-### Usage on a Cisco device
+by the included minitest suite).  The Client and Node APIs will read
+connection information (host, username, etc.) from a
+[configuration file](#configuration). When creating a Client or Node,
+you can specify the label of the node in the config file to connect to (if no
+label is specified, "default" is assumed). If a configuration file cannot be
+found, the Client or Node will attempt to connect to the local device.
 
 #### Low-level Client API
 
 ```ruby
 require 'cisco_node_utils'
 
-# get a connection to the local device
+# get a connection to the "default" node in the config file
 client = Cisco::Client.create()
 
-client.get(command: 'show version')
+# get a connection to the "nxapi_remote" node in the config file
+#client = Cisco::Client.create('nxapi_remote')
+
+# via cli
+version = client.get(command: 'show version')
 client.set(values: 'vtp domain mycompany.com')
+
+# via yang
+bgp_cfg = client.get(data_format: :yang_json, command: '{"Cisco-IOS-XR-ipv4-bgp-cfg:bgp": [null]}')
+client.set(data_format: :yang_json, 
+           values: '{"Cisco-IOS-XR-infra-rsi-cfg:vrfs":{"vrf":[{"vrf-name":"BLUE","create":[null]}]}}')
 ```
 
 #### High-level Node API
 
 ```ruby
 require 'cisco_node_utils'
 
-# get a connection to the local device
+# get a connection to the "default" node in the config file
 node = Cisco::Node.instance()
 
-version = node.config_get("show_version", "system_image")
-
-node.config_set("vtp", "domain", "mycompany.com")
-```
-
-### Remote usage
-
-#### Low-level Client API
-
-```ruby
-require 'cisco_node_utils'
-
-client = Cisco::Client.create('n3k.mycompany.com', 'username', 'password')
```

Comment:
It's a good thing that you went through this section and discovered the outdated/wrong information!  Here are some thoughts on the content and organization.

- I see that you have consolidated everything under the `Low-level Client API` and `High-level Client API` sections but for clarity I don't want to loose the usage context headings: `Usage on a Cisco device` vs. `Remote usage`.  With the re-write, IMO this distinction is not as clear.
- There was some additional wrong information  for NXOS corrected in my suggested update below.
- I would also like to massage the existing nxos examples so that enough context is provided to be useful.
- I also feel like we should clarify nxos vs xr examples.

How about this:

### Cisco Nexus Device (API's Supported: CLI)

#### Low-level Client API
```ruby
require 'cisco_node_utils'

######################################
# Local Device Usage:                #
######################################

# Create connection.
client = Cisco::Client.create()

# Use connection to get and set device state.
client.set(values: 'feature vtp')
client.set(values: 'vtp domain mycompany.com')
version = client.get(command: 'show vtp status | inc Domain')

######################################
# Remote Device Usage:               #
######################################

# Create a connection from a remote server to the following:
# - 'default' device defined in the cisco_node_utils.yaml file on the remote server.
# - 'n9k' device defined in the cisco_node_utils.yaml file on the remote server.
client1 = Cisco::Client.create()
client2 = Cisco::Client.create('n9k')
# Warning: Make sure to exclude devices using the 'no_proxy' environment variable
# to ensure successful remote connections.

# Use connection to get and set device state.
client1.set(values: 'feature vtp')
client1.set(values: 'vtp domain mycompany.com')
version = client1.get(command: 'show vtp status | inc Domain')

client2.get(command: 'show version')
```
#### High-level Client API
```ruby
require 'cisco_node_utils'

######################################
# Local Device Usage:                #
######################################

# Create connection.
node = Cisco::Node.instance()

# Use connection to get and set device state.
node.config_set('feature', 'vtp', state: '')
node.config_set('vtp', 'domain', domain: 'mycompany.com')
version = node.config_get('vtp', 'domain')

######################################
# Remote Device Usage:               #
######################################

# Create a connection from a remote server to the following:
# - 'default' device defined in the cisco_node_utils.yaml file on the remote server.
node = Cisco::Node.instance()
# OR:
# - 'n9k' device defined in the cisco_node_utils.yaml file on the remote server.
# Cisco::Environment.default_environment_name = 'n9k'
# node = Cisco::Node.instance()

# Warning: Make sure to exclude devices using the 'no_proxy' environment variable
# to ensure successful remote connections.

# Use connection to get and set device state.
node1.config_set('feature', 'vtp', state: '')
node1.config_set('vtp', 'domain', domain: 'mycompany.com')
version = node1.config_get('vtp', 'domain')
```

### Cisco XR Device (API's Supported: CLI, YANG)

#### Low-level Client API
```
CLI and YANG examples here
```
#### High-level Client API
```
CLI and YANG examples here
```